packages/libs/pactjs-generator/src/lexer.ts

```diff
@@ -0,0 +1,266 @@
+import type { Lexer } from 'moo';
+import * as moo from 'moo';
+
+/**
+ * @internal
+ */
+export const lexer: Lexer = moo.compile({
+  number: /-?(?:0|[1-9]\d*)(?:\.\d+)?(?:[eE][+-]?\d+)?/,
+  // https://regex101.com/r/JmRAEN/1
+  TYPED_IDEN:
+    /(?:[a-zA-Z_0-9]+(?:-)*[a-zA-Z_0-9]+)+:(?:[a-zA-Z_{}.0-9]+(?:-)*)+[a-zA-Z_{}.0-9]*/,
+  // https://regex101.com/r/CKdphk/1
+  name: {
+    match: /(?:[a-zA-Z_0-9*]+(?:-)*)+[a-zA-Z_0-9]*\*?/,
+    type: moo.keywords({
+      KW_DEFUN: 'defun',
+      KW_MODULE: 'module',
+    }),
+  },
+  postEscapedNewline: /\s*[\\|\\\\]/,
+  WS: /[ \t]+/,
+  lparen: '(',
+  rparen: ')',
+  lsquare: '[',
+  rsquare: ']',
+  lcurly: '{',
+  rcurly: '}',
+  assign: ':=',
+  comma: ',',
+  string: { match: /"(?:[^"\\]|\\.|\\\n)*"/, lineBreaks: true },
+  newlineEscape: {
+    match: /[\\|\\\\][\n]\s*/,
+    lineBreaks: true,
+  },
+  singleQuote: /'/,
+  doubleQuote: /"/,
+  plus: '+',
+  colon: ':',
+  equals: '=',
+  dash: '-',
+  comment: /;;?.*/,
+  module: /\'[a-zA-Z_]+/,
+  guard: /@[a-zA-Z_]+/,
+  gt: '>',
+  lt: '<',
+  NOT_IS: /!=/,
+  NL: { match: /\n/, lineBreaks: true },
+});
+
+/**
+ * @internal
+ */
+export function flattenRecursive(arr: any[]): unknown[] {
+  return arr.reduce((acc, cur) => {
+    if (Array.isArray(cur)) {
+      return acc.concat(flattenRecursive(cur));
+    }
+    return acc.concat(cur);
+  }, []);
+}
+
+/**
+ * @internal
+ */
+export const flattenOnce = (arr: any[]) => {
+  return arr.reduce((acc, cur) => {
+    if (Array.isArray(cur)) {
```

Comment:
Do we need this check as it does exactly the same as 68

Reply:
Wdym?